pkg/actuators/machine/reconciler.go

```diff
@@ -0,0 +1,363 @@
+package machine
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	machinecontroller "github.com/openshift/machine-api-operator/pkg/controller/machine"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	errorutil "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog"
+
+	awsproviderv1 "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsprovider/v1beta1"
+)
+
+const (
+	requeueAfterSeconds      = 20
+	requeueAfterFatalSeconds = 180
+)
+
+// Reconciler runs the logic to reconciles a machine resource towards its desired state
+type Reconciler struct {
+	*machineScope
+}
+
+func newReconciler(scope *machineScope) *Reconciler {
+	return &Reconciler{
+		machineScope: scope,
+	}
+}
+
+// create creates machine if it does not exists.
+func (r *Reconciler) create() error {
+	klog.Infof("%s: creating machine", r.machine.Name)
+
+	if err := validateMachine(*r.machine); err != nil {
+		return fmt.Errorf("%v: failed validating machine provider spec: %v", r.machine.GetName(), err)
+	}
+
+	// We explicitly do NOT want to remove stopped masters.
+	isMaster, err := r.isMaster()
+
```

Comment:
Nit, would drop this empty new line to pair the `if err` with the call to `isMaster`